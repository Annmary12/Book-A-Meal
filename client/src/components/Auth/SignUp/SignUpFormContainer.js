import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { signUpSchema } from '../../../utils/validation-schemas/authInfoSchemas.js';
import AuthForm from '../AuthForm.js';
import { signup } from '../../../actions/auth.js';

const signUpFormConfig = {
  mapPropsToValues: () => ({
    fullName: '',
    username: '',
    email: '',
    role: '',
    password: '',
    passwordConfirm: ''
  }),
  validationSchema: signUpSchema,
  handleSubmit: async (values, { setSubmitting, props }) => {
    setSubmitting(true);
    await props.dispatch(signup(values));
    setSubmitting(false);
  }
};
const mapStateToProps = () => ({
  type: 'signup'
});

export default compose(
  connect(mapStateToProps),
  withFormik(signUpFormConfig)
)(AuthForm);
