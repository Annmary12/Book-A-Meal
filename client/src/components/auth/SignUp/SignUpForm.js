import React from 'react';
// import PropTypes from 'prop-types';
import Loader from '../../Loader.js';
import styles from '../auth.scss';

const SignUpForm = ({
  values,
  touched,
  errors,
  dirty,
  isSubmitting,
  handleChange,
  handleBlur,
  handleSubmit,
  handleReset
}) => (
  <form onSubmit={handleSubmit}>
    <label className={styles.label} htmlFor="fullName">Full name:
      <input
        className={
          errors.fullName && touched.fullName
            ? styles.inputError : styles.input}
        type="text"
        name="fullName"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.fullName}
      />
      {errors.fullName &&
      touched.fullName && <div className={styles.errorMessage}>{errors.fullName}</div>}
    </label>

    <label className={styles.label} htmlFor="username">Username:
      <input
        className={
          errors.username && touched.username
            ? styles.inputError : styles.input}
        type="text"
        name="username"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.username}
      />
      {errors.username &&
      touched.username && <div className={styles.errorMessage}>{errors.username}</div>}
    </label>

    <label className={styles.label} htmlFor="email">Email:
      <input
        className={
          errors.email && touched.email
            ? styles.inputError : styles.input}
        type="email"
        name="email"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.email}
      />
      {errors.email &&
      touched.email && <div className={styles.errorMessage}>{errors.email}</div>}
    </label>

    <label className={styles.label} htmlFor="role">Role:
      <select
        className={
          errors.role && touched.role
            ? styles.selectError : styles.select}
        name="role"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.role}
      >
        <option value="" disabled>Select a role</option>
        <option value="customer">Customer</option>
        <option value="caterer">Caterer</option>
      </select>
      {errors.role &&
      touched.role && <div className={styles.errorMessage}>{errors.role}</div>}
    </label>

    <label className={styles.label} htmlFor="password">Password:
      <input
        className={
          errors.password && touched.password
            ? styles.inputError : styles.input}
        type="password"
        name="password"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.password}
      />
      {errors.password &&
      touched.password && <div className={styles.errorMessage}>{errors.password}</div>}
    </label>

    <label className={styles.label} htmlFor="password-confirmation">Confirm password:
      <input
        className={
          errors.passwordConfirm && touched.passwordConfirm
            ? styles.inputError : styles.input}
        type="password"
        name="passwordConfirm"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.passwordConfirm}
      />
      {errors.passwordConfirm &&
      touched.passwordConfirm && <div className={styles.errorMessage}>{errors.passwordConfirm}</div>}
    </label>

    <button
      className={isSubmitting ? styles.loginBtnSubmit : styles.loginBtn}
      type="button"
      onClick={handleSubmit}
      disabled={isSubmitting}
    >
      {isSubmitting ? <Loader message="" width={30} /> : 'Sign Up'}
    </button>
  </form>
);

export default SignUpForm;
