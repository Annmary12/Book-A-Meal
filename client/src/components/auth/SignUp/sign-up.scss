.title{
  font-family: 'Dancing Script', cursive;
  font-size: 40px;
  margin-top: 0px;
}