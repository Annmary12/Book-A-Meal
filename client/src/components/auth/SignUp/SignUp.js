import React from 'react';
import { Link } from 'react-router-dom';
import AuthArea from '../AuthArea.js';
import SignUpFormContainer from './SignUpFormContainer.js';

const SignUp = () => (
  <AuthArea>
    <SignUpFormContainer />
    <p style={{ textAlign: 'center' }}>
      Already have an account? <Link to="login">Login</Link>
    </p>
  </AuthArea>
);

export default SignUp;
