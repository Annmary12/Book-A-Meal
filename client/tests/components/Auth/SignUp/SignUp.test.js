import React from 'react';
import SignUp from '../../../../src/components/Auth/SignUp/SignUp';

describe('SignUp component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SignUp />);

    expect(wrapper).toMatchSnapshot();
  });
});