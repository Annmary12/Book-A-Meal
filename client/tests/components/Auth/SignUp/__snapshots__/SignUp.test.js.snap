// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should match snapshot 1`] = `
<AuthArea>
  <Connect(C) />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Already have an account? 
    <Link
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </p>
</AuthArea>
`;
